backend/mysql_utils.js

```diff
@@ -0,0 +1,228 @@
+                        }
+
+                    }
+                    return callback(null, { grants });
+                }
+            });
+        } else {
+            return dbError(err, callback);
+        }
+    });
+}
+
+
+exports.storeRecord = function (userId, recordName, baseUrl, recordId, mysqlConnPool) {
+    const sql = `INSERT INTO records(id, name, creator, link) VALUES ('${recordId}', '${recordName}','${userId}', '${baseUrl}')`;
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, function (err, result) {
+                conn.release();
+                if (err) {
+                    console.error("error storing record link")
+                }
+            });
+        }
+    });
+}
+
+exports.deleteRecord = function (recordId, mysqlConnPool) {
+    const sql = `DELETE FROM records WHERE id='${recordId}'`;
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, function (err, result) {
+                conn.release();
+                if (err) {
+                    console.error(`error deleting record link: ${err}`)
+                }
+            });
+        }
+    });
+};
+
+exports.userHavePrivileges = function (recordId, userId, mysqlConnPool, callback) {
+    const sql = `SELECT * FROM records WHERE id='${recordId}' and creator='${userId}'`;
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, function (err, result) {
+                conn.release();
+                if (err) {
+                    return dbError(err, callback)
+                }
+                if (result.length) {
+                    return callback(null, true);
+                } else {
+                    return callback(null, false);
+                }
+            });
+        } else {
+            return dbError(err, callback);
+        }
+    });
+}
+
+
+exports.getRecords = function (userId, isAdmin, mysqlConnPool, callback) {
+    let sql = "SELECT records.id, records.creator, records.name,  DATE_FORMAT(records.publish_date, '%Y-%m-%d %r') publish_date, records.link, users.email_id FROM records LEFT JOIN users ON records.creator=users.id";
+    if (!isAdmin) sql += ` WHERE creator='${userId}'`;
+
+    mysqlConnPool.getConnection(function (err, conn) {
+        if (!err) {
+            conn.query(sql, function (err, results) {
+                conn.release();
+                if (!err) {
+                    let records = {};
+                    if (results.length) {
+                        results.forEach(function (item, index) {
+                            records[item.id] = {
+                                id: item.id,
+                                name: item.name,
+                                publishDate: item.publish_date,
+                                link: item.link,
+                                creator: (item.email_id) ? item.email_id : item.creator,
```

Comment:
```suggestion
                                creator: item.email_id ? item.email_id : item.creator,
```

Reply:
i think its good for my understanding

Reply:
I think it looks better the way its implemented. its better for understanding.